### ISSUE TITLE
ApolloEngineOptions should be included in CodegenConfig.schema

### ISSUE THREAD

**Issue by n8downs** [dotansimha/graphql-code-generator#8307]
### Describe the bug

I'm trying to use the new typescript configs in order to eliminate duplication between our prod and dev configs. However, the [apollo-engine](https://www.the-guild.dev/graphql/codegen/docs/config-reference/schema-field#apollo-engine) schema we're using seems to cause a type error:
```
error TS2322: Type '{ engine: { apiKey: string; }; graph: string; variant: string; }' is not assignable to type 'SchemaFromCodeFileOptions'.
  Object literal may only specify known properties, and 'engine' does not exist in type 'SchemaFromCodeFileOptions'.

 9         engine: {
           ~~~~~~~~~
10           apiKey: '....',
   ~~~~~~~~~~~~~~~
11         },
   ~~~~~~~~~
```

Digging through the types, it appears to me that `ApolloEngineOptions` from `@graphql-tools/apollo-engine-loader` should be included in `Types.Schema` from `@graphql-codegen/plugin-helpers`

### Your Example Website or App

sorry

### Steps to Reproduce the Bug or Issue

Include an apollo-engine config as a schema
Run codegen

### Expected behavior

I expect code to be generated based on my apollo-engine schema

### Platform

- OS: macOS
- NodeJS: 18.0.0
- `graphql` version: 16.5.0
- `@graphql-codegen/cli` version(s): 2.12.0

### Codegen Config File

```
import baseConfig from './codegen.base';
import { CodegenConfig } from '@graphql-codegen/cli';

const config: CodegenConfig = {
  ...baseConfig,
  schema: {
    'apollo-engine': {
      engine: {
        apiKey: '...',
      },
      graph: '...',
      variant: 'prod',
    },
  },
};

export default config;
```

**Comment by charlypoly** [comment]
Hi @n8downs,

you are right!

Could you try again with the following alpha release?

```
@graphql-codegen/cli@2.12.1-alpha-20220920093645-344b6e55d
```

I tested on my side with success ✅

**Comment by charlypoly** [comment]
Fixed in `@graphql-codegen/cli@2.12.1` 📦
